### PR TITLE
Implement terraform for deploying to CH Development

### PR DESCRIPTION
**Pull request by JoelClemence**
* Creates two fargate ECS Tasks - server/frontend
* Deploys Postgres RDS instance
* Add ALB and Route53 domain/ACM Certificate

### CODE REVIEW

File: groups/infrastructure/alb.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module "alb" {
+  source = "git@github.com:companieshouse/terraform-modules//aws/application_load_balancer?ref=1.0.205"
+
+  environment                = var.environment
+  service                    = local.service_name
+  ssl_certificate_arn        = aws_acm_certificate.certificate.arn
+  subnet_ids                 = data.aws_subnets.public.ids
```
Comment (ch-tstrickland) [inline groups/infrastructure/alb.tf:11]: Should this be private.ids?

File: groups/infrastructure/ecr.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+resource "aws_ecr_lifecycle_policy" "lifecycle" {
+  repository = aws_ecr_repository.proxy_sidecar.name
+  policy     = <<EOF
+{
+    "rules": [
+        {
+            "rulePriority": 1,
+            "description": "Expire images older than 7 days",
```
Comment (ch-tstrickland) [inline groups/infrastructure/ecr.tf]: Expire untagged images older than 7 days

File: groups/infrastructure/iam.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:BatchGetImage",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "ssmmessages:CreateControlChannel",
+        "ssmmessages:CreateDataChannel",
+        "ssmmessages:OpenControlChannel",
+        "ssmmessages:OpenDataChannel",
+        "elasticfilesystem:*"
```
Comment (ch-tstrickland) [inline groups/infrastructure/iam.tf:59]: ^^ Comments on unusual fields to help maintainers/reviewers

**Review by ch-tstrickland — COMMENTED** [review]
A comment at the top of each file stating function

File: groups/api/ecs.tf [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  docker_registry   = var.docker_registry
+  docker_repo       = local.server_requirements.docker_registry
+  container_version = var.applications_developer_version
+  container_port    = local.container_port
+
+  # Service configuration
+  service_name = local.service_name
+  name_prefix  = local.service_name
+
+  # Service Healthcheck configuration
```
Comment (ch-tstrickland) [inline groups/api/ecs.tf]: Delete comment?

File: groups/infrastructure/alb.tf (hunk 1 above)
Comment (JoelClemence) [reply groups/infrastructure/alb.tf:11]: No this should be public. I think it is quite a confusing setup but this follows the pattern set by other ALBs. There are no publicly accessible ALBs in private subnets.

**Review by ch-tstrickland — APPROVED** [review]

**Comment by ch-tstrickland** [comment]
I do not know Terraform in depth, but I have walked through this with Joel and am happy that it runs happily on fair weather paths & that I've helped Joel reflect on the changes.

**Pull request merged**